source/extensions/filters/http/dynamic_forward_proxy/proxy_filter.cc

```diff
@@ -317,20 +323,33 @@ Http::FilterHeadersStatus ProxyFilter::decodeHeaders(Http::RequestHeaderMap& hea
     }
   }
 
-  // Get host value from the request headers.
+  // Parse the authority to get host and port information.
   const auto host_attributes =
       Http::Utility::parseAuthority(headers.Host()->value().getStringView());
-  absl::string_view host = host_attributes.host_;
```

Comment:
I fixed this internally by putting back the brackets for IPv6 literals.

```suggestion
  std::string host_str(!host_attributes.is_ip_address_ || !absl::StrContains(host_attributes.host_, ":") ? host_attributes.host_ : absl::StrCat("[", host_attributes.host_, "]"));
  absl::string_view host = host_str;
```

This is the smallest fix without the need to deal with the rest of the code.